### DISCUSSION TITLE
Should we think of the whole thing as a Rust-first cosmos-sdk alternative?

### DISCUSSION THREAD

**Discussion by patrick-ogrady** [commonwarexyz/monorepo#232] [Q&A] [answered]
<img width="602" alt="image" src="https://github.com/user-attachments/assets/47f3a741-5d5e-40ca-9ab8-3f4031d21482" />

Source: https://x.com/mhluongo/status/1821635684849090772?s=46&t=eGRNsJ9UUBdKbeXgtcMh9A

**Comment by patrick-ogrady** [comment, marked as answer]
TL;DR yes and no!

Like the Cosmos-SDK, the Commonware Framework (once further along) will enable a developer to launch their own blockchain.

Unlike the Cosmos-SDK, the Commonware Framework (CF) will not wrap the underlying network, consensus, and storage primitives behind a unified interface (like ABCI) nor will it define a standard block, transaction, or state/trie format. Everything released by Commonware will be built in Rust and should be easy to extend/modify with your Rust (unless I really fuck something up lol). The primitives produced by Commonware will predominantly be tailored for throughput and will make unique tradeoffs.

There are some new complexities that come from this approach (which we’ll address in the coming months) but the idea in a nutshell is that developers will commonly “specialize” their stack for an application by composing primitives provided by Commonware and some of their own (especially on the execution layer) via layers of well-defined interfaces.

All this being said, Commonware clearly has a long way to go and the Cosmos-SDK has attracted incredible mindshare over the last few years with their approach (me included). There are definitely a large number of talented developers who prefer their abstraction and the tradeoffs they’ve made.

Looking forward to getting your feedback as we release more primitives!

Source: https://x.com/_patrickogrady/status/1821722302830276991?s=46&t=eGRNsJ9UUBdKbeXgtcMh9A

**Reply by patrick-ogrady** [reply]
I think over time we'll produce different "dialects" of the same primitive (i.e. fully-connected p2p vs sparsely-connected p2p) based on the needs of the applications developers want to launch.

However, I think the short/medium-term will be focused on democratizing access to high capacity.

Source: https://x.com/_patrickogrady/status/1821723453646631402?s=46&t=eGRNsJ9UUBdKbeXgtcMh9A
